debezium-core/src/main/java/io/debezium/relational/history/FileDatabaseHistory.java

```diff
@@ -78,7 +78,11 @@ public void start() {
                     if (!storageExists()) {
                         // Create parent directories if we have them ...
                         if (path.getParent() != null) {
-                            Files.createDirectories(path.getParent());
+                            try {
+                                Files.createDirectories(path.getParent());
```

Comment:
So after thinking more about this: did you actually encounter any issue here, or is it rather a theoretical concern? I'm asking because `createDirectories()` doesn't raise `FileAlreadyExistsException` if any parent dirs exist. It will raise an exception though if the parent path denotes a file, which makes sense. So in that light I'm inclined to say everything works here as it should. WDYT?

Reply:
When i set the filename: /tmp/history.dat(not exist) in my Mac OS with common user(not root), it does raise the exception.

Reply:
Could you share the stacktrace, if you still have it?

Reply:
```shell
io.debezium.relational.history.DatabaseHistoryException: Unable to create history file at /tmp/history.dat: /tmp
	at io.debezium.relational.history.FileDatabaseHistory.lambda$start$0(FileDatabaseHistory.java:92) ~[debezium-core-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.util.FunctionalReadWriteLock.write(FunctionalReadWriteLock.java:99) ~[debezium-core-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.relational.history.FileDatabaseHistory.start(FileDatabaseHistory.java:69) ~[debezium-core-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.relational.HistorizedRelationalDatabaseSchema.<init>(HistorizedRelationalDatabaseSchema.java:40) ~[debezium-core-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.connector.oracle.OracleDatabaseSchema.<init>(OracleDatabaseSchema.java:37) ~[debezium-connector-oracle-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.connector.oracle.OracleConnectorTask.start(OracleConnectorTask.java:57) ~[debezium-connector-oracle-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.connector.common.BaseSourceTask.start(BaseSourceTask.java:130) ~[debezium-core-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.embedded.EmbeddedEngine.run(EmbeddedEngine.java:759) ~[debezium-embedded-1.6.1.Final.jar:1.6.1.Final]
	at io.debezium.embedded.ConvertingEngineBuilder$2.run(ConvertingEngineBuilder.java:188) ~[debezium-embedded-1.6.1.Final.jar:1.6.1.Final]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_201]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_201]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_201]
Caused by: java.nio.file.FileAlreadyExistsException: /tmp
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:88) ~[na:1.8.0_201]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102) ~[na:1.8.0_201]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107) ~[na:1.8.0_201]
	at sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:384) ~[na:1.8.0_201]
	at java.nio.file.Files.createDirectory(Files.java:674) ~[na:1.8.0_201]
	at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781) ~[na:1.8.0_201]
	at java.nio.file.Files.createDirectories(Files.java:727) ~[na:1.8.0_201]
	at io.debezium.relational.history.FileDatabaseHistory.lambda$start$0(FileDatabaseHistory.java:81) ~[debezium-core-1.6.1.Final.jar:1.6.1.Final]
	... 11 common frames omitted
```

Reply:
@gunnarmorling Sorry for my later reply.

Reply:
Hum, hum; that's really odd. Are you sure that `/tmp` actually is a directory and not a file?

Reply:
omg. It's a symlink.🤭

Reply:
Oook, that helps :) Symlink pointing to a directory or a file?

Reply:
That's a directory.